test/evaluation/test_eval_utils.py

```diff
@@ -0,0 +1,244 @@
+
+    assert output == deserialized_output
+
+
+def test_convert_list_strings_output():
+    """
+    Test serialization of outputs to dictionary and deserialization to dataclass  for a component that returns a list of strings.
+    """
+    generator = HuggingFaceLocalGenerator(
+        model_name_or_path="google/flan-t5-small",
+        task="text2text-generation",
+        generation_kwargs={"max_new_tokens": 100, "temperature": 0.1, "do_sample": True},
+    )
+    generator.warm_up()
+    output = generator.run(
+        "Context: David Culpepper is the best American actor. \n Who is the best American actor? \n Answer: David Culpepper"
+    )
+    dict_serialized_output = convert_component_output_to_dict(output)
+    deserialized_output = convert_component_output_from_dict(dict_serialized_output)
+
+    assert output == deserialized_output
+
+
+def test_convert_list_generated_answers_outputs():
+    """
+    Test serialization of outputs to dictionary and deserialization to dataclass  for a component that returns a list of Generated Answers.
+    """
+    documents = [
+        Document(content="My name is Jean and I live in Paris."),
+        Document(content="My name is Mark and I live in Berlin."),
+        Document(content="My name is Giorgio and I live in Rome."),
+    ]
+
+    answer_builder = AnswerBuilder()
+    output = answer_builder.run(query="Who lives in Paris?", replies=["Jean"], metadata=[{}], documents=documents)
+    dict_serialized_output = convert_component_output_to_dict(output)
+    deserialized_output = convert_component_output_from_dict(dict_serialized_output)
+
+    assert output == deserialized_output
+
+
+def test_convert_component_list_documents_outputs():
+    """
+    Test serialization of multiple component outputs to dictionary and deserialization to dataclass for a component that returns a list of Documents.
+    """
+    documents = [
+        Document(content="My name is Jean and I live in Paris."),
+        Document(content="My name is Mark and I live in Berlin."),
+        Document(content="My name is Giorgio and I live in Rome."),
+    ]
+    document_store = InMemoryDocumentStore()
+    document_store.write_documents(documents)
+    retriever = InMemoryBM25Retriever(document_store=document_store)
+    inputs = ["Who lives in Paris?", "Who lives in Berlin?", "Who lives in Rome?"]
+    outputs = []
+    for input in inputs:
+        outputs.append(retriever.run(input))
+    dict_serialized_outputs = convert_component_outputs_to_dict(outputs)
+    deserialized_outputs = convert_component_outputs_from_dict(dict_serialized_outputs)
+
+    assert outputs == deserialized_outputs
+
+
+def test_convert_component_list_strings_outputs():
+    """
+    Test serialization of multiple component outputs to dictionary and deserialization to dataclass for a component that returns a list of strings.
+    """
+    generator = HuggingFaceLocalGenerator(
```

Comment:
Same as above: let's not use a model in these unit tests. It slows them down drastically.